icu4c/source/i18n/timezone.cpp

```diff
@@ -1497,8 +1497,10 @@ TimeZone::hasSameRules(const TimeZone& other) const
 static void U_CALLCONV initTZDataVersion(UErrorCode &status) {
     ucln_i18n_registerCleanup(UCLN_I18N_TIMEZONE, timeZone_cleanup);
     int32_t len = 0;
-    UResourceBundle *bundle = ures_openDirect(NULL, kZONEINFO, &status);
-    const UChar *tzver = ures_getStringByKey(bundle, kTZVERSION, &len, &status);
+    UResourceBundle bundle;
+    ures_initStackObject(&bundle);
+    ures_openDirectFillIn(&bundle, NULL, kZONEINFO, &status);
```

Comment:
Does this intentionally not check for success of the status before making the next call?

Reply:
Functions that take UErrorCode can be (and usually are) "chained". Only when you start calling something that depends on the output of such a function but does not take UErrorCode (or dereferencing a pointer returned from one, etc.) do you have to check.

http://userguide.icu-project.org/dev/codingguidelines#TOC-Details-about-ICU-Error-Codes

Reply:
So in this case, the `initTZDataVersion` function is called by `umtx_initOnce`, which checks for `U_FAILURE(status)`.
This means that at the beginning of this function (`initTZDataVersion`) status will always be U_SUCCESS.